### PR TITLE
fix(buttons): add the type of bare in button-icon.

### PR DESCRIPTION
**Pull request by esarbanis**
We need the bare type icons for cases such as the sorting icon in [data tables](https://www.lightningdesignsystem.com/components/data-tables/)

### CODE REVIEW

**Comment by bekos** [comment]
Probably their datatable examples are not updated yet. Based on the [1.0.0 changelog](https://www.lightningdesignsystem.com/release-notes/) the `.slds-button--icon-bare` is deprecated and you should use `.slds-button--icon` instead. Does this work for you?

**Comment by esarbanis** [comment]
I cannot achieve this since it will default to [`.slds-button--icon-border`](https://github.com/ng-lightning/ng-lightning/blob/master/src/buttons/button-icon.ts#L13)

Should I go ahead and change the default?

**Comment by bekos** [comment]
@esarbanis I see. I would prefer to provide a way to bypass the default method if `nglButtonIcon` is empty, ie `nglButtonIcon=""`. Do you want to take a stub at this?

**Comment by esarbanis** [comment]
Already on it :)

Will also introduce a separate spec for button-icon

**Comment by esarbanis** [comment]
I have send an initial study for 'bare' button icons for review.

File: src/buttons/button-icon.spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import {NglButtonIcon} from './button-icon';
+
+function getButtonElement(element: Element): HTMLButtonElement {
+  return <HTMLButtonElement>element.querySelector('button');
+}
+
+describe('`nglButtonIcon`', () => {
+
+  it('should render the default button icon when the type is empty', testAsync(
+    `<button [nglButtonIcon]="style">Go <ngl-icon-button icon="right"></ngl-icon-button></button>`, ({fixture, done}) => {
```
Comment (bekos) [inline src/buttons/button-icon.spec.ts]: @esarbanis No need to have `<ngl-icon-button icon="right"></ngl-icon-button>` in your HTML, as we don't test this.

File: src/buttons/button-icon.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  });
+}
+
+@Component({
+  directives: [NglButtonIcon],
+  template: '',
+})
+export class TestComponent {
+  style: string = '';
+  size: string;
```
Comment (bekos) [inline src/buttons/button-icon.spec.ts]: @esarbanis Size is not used.

File: src/buttons/button-icon.ts [hunk 3]
```diff
@@ -18,4 +19,9 @@ export class NglButtonIcon {
     this.renderer.setElementClass(this.element.nativeElement, 'slds-button', true);
   }
 
-};
+  private normalize(type: string): string {
+    if (type == null) return '-border';
```
Comment (bekos) [inline src/buttons/button-icon.ts]: @esarbanis Can you just `if` it instead of comparing it with `null`. I suppose we want to handle all cases like `null`, `undefined`, `false` or empty string the same way. Is it correct?

**Comment by bekos** [comment]
@esarbanis Very good! Because we are dealing with edge cases I would like to have more explicit edge test cases, like `<button nglButtonIcon>` and `<button nglButtonIcon="">`. Can u add these?

File: src/buttons/button-icon.ts (hunk 3 above)
Comment (esarbanis) [inline src/buttons/button-icon.ts]: Actually we need to leave out the empty string case, that's what the double equals is for.

**Comment by bekos** [comment]
Very good job @esarbanis! Thx!
